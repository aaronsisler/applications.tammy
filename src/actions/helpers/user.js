export const setUser = (user) => ({
    type: 'SET_USER',
    user,
});

export const editUser = (updates) => ({
    type: 'EDIT_USER',
    updates,
});

export const clearUser = () => ({
    type: 'CLEAR_USER'
});
