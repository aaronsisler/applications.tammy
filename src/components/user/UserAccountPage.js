import React from 'react';
import UserProfileContainer from './UserProfileContainer';

export default class UserAccountPage extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
    }

    render() {
        return (
            <div id="user_account_page">
                <UserProfileContainer />
            </div>
        );
    }
}
