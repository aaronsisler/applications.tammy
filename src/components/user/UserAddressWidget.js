import React from 'react';
import PropTypes from 'prop-types';
import InputTools from 'User/tools/inputs';

export default class UserAddressWidget extends React.Component {
    inputTools = new InputTools();

    constructor(props) {
        super(props);
        this.state = {
            addressLine1: props.user.addressLine1 ? props.user.addressLine1 : '',
            addressLine2: props.user.addressLine2 ? props.user.addressLine2 : '',
            city: props.user.city ? props.user.city : '',
            state: props.user.state ? props.user.state : '',
            postalCode: props.user.postalCode ? props.user.postalCode : '',
        };
    }

    componentDidUpdate(prevProps) {
        const { user: oldUser } = prevProps;
        const { user: newUser } = this.props;

        if (oldUser.userId !== newUser.userId) {
            return this.setState(() => ({
                addressLine1: newUser.addressLine1 ? newUser.addressLine1 : '',
                addressLine2: newUser.addressLine2 ? newUser.addressLine2 : '',
                city: newUser.city ? newUser.city : '',
                state: newUser.state ? newUser.state : '',
                postalCode: newUser.postalCode ? newUser.postalCode : '',
            }));
        }
    }

    handleInputChange = (e) => {
        const inputName = e.target.name;
        const inputValue = e.target.value;

        switch (inputName) {
            case 'postalCode':
                this.handlePostalCodeValidation(inputValue);
                return;
            case 'state':
                this.handleStateAbbreviationValidation(inputValue);
                return;
            default:
                this.handleAddInput(inputName, inputValue);
                return;
        }
    }

    handlePostalCodeValidation = (inputValue) => {
        if (!inputValue || inputValue.match(/^\d{1,5}?$/)) {
            this.setState(() => ({ postalCode: inputValue }));
        }
    }

    handleStateAbbreviationValidation = (inputValue) => {
        if (!inputValue || inputValue.match(/^[A-Za-z]{1,2}$/)) {
            this.setState(() => ({ state: inputValue.toUpperCase() }));
        }
    }

    handleAddInput = (inputName, inputValue) => (
        this.setState(() => ({
            [inputName]: inputValue
        }))
    )

    handleSubmit = () => {
        const submitObject = Object.assign({}, this.state);
        this.props.onSubmit(submitObject);
    }

    render() {
        return (
            <div className="user_address_widget">
                <div className="user_address_content">
                    <div className="user_address_widget__title">
                        Address
                    </div>
                    <div className="user_address_widget__inputs">
                        <input
                            readOnly={this.props.isReadOnly}
                            type="text"
                            name="addressLine1"
                            placeholder="Address Line 1"
                            value={this.state.addressLine1}
                            onChange={this.handleInputChange}
                            onBlur={this.inputTools.handleRequiredValidation}
                        />
                        <input
                            readOnly={this.props.isReadOnly}
                            type="text"
                            name="addressLine2"
                            placeholder="Address Line 2"
                            value={this.state.addressLine2}
                            onChange={this.handleInputChange}
                        />
                        <div className="user_address_widget__inputs_location">
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="city"
                                placeholder="City"
                                value={this.state.city}
                                onChange={this.handleInputChange}
                                onBlur={this.inputTools.handleRequiredValidation}
                            />
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="state"
                                placeholder="State"
                                value={this.state.state}
                                onChange={this.handleInputChange}
                                onBlur={this.inputTools.handleRequiredValidation}
                            />
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="postalCode"
                                placeholder="Postal Code"
                                value={this.state.postalCode}
                                onChange={this.handleInputChange}
                                onBlur={this.inputTools.handleRequiredValidation}
                            />
                        </div>
                    </div>
                </div>
                {!this.props.isReadOnly &&
                    <div className="user_address_widget__button">
                        <button
                            disabled={!this.state.addressLine1 || !this.state.city || !this.state.state || !this.state.postalCode}
                            onClick={this.handleSubmit}
                        >
                            Update Address
                       </button>
                    </div>
                }
            </div>
        )
    }
}

UserAddressWidget.propTypes = {
    isReadOnly: PropTypes.bool,
    user: PropTypes.object.isRequired,
    onSubmit: PropTypes.func,
};
