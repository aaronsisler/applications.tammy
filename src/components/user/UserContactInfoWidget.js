import React from 'react';
import PropTypes from 'prop-types';
import { requiredInputFieldClassName } from '../../tools/constants';
import { isEmailValid } from '../../tools/email';

export default class UserNameWidget extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isEmailValid: props.user.email ? true : false,
            email: props.user.email ? props.user.email : '',
            phoneAreaCode: props.user.phoneAreaCode ? props.user.phoneAreaCode : '',
            phonePrefix: props.user.phonePrefix ? props.user.phonePrefix : '',
            phoneLineNumber: props.user.phoneLineNumber ? props.user.phoneLineNumber : '',
            phoneExt: props.user.phoneExt ? props.user.phoneExt : '',
        };
    }

    handleEmailInput = (e) => {
        const inputValue = e.target.value;

        this.setState({ email: inputValue });
    }

    handleEmailValidation = (e) => {
        const inputName = e.target.name;
        const inputValue = e.target.value;

        if (isEmailValid(inputValue)) {
            document.getElementsByName(inputName)[0].classList.remove(requiredInputFieldClassName);
            return this.setState({ isEmailValid: true });
        }
        else {
            document.getElementsByName(inputName)[0].classList.add(requiredInputFieldClassName);
            return this.setState({ isEmailValid: false });
        }
    }

    handlePhoneNumberValidation = (e) => {
        const inputName = e.target.name;
        const inputValue = e.target.value;

        if (!inputValue) {
            return this.setState({ [inputName]: inputValue });
        }

        switch (inputName) {
            case 'phoneAreaCode':
            case 'phonePrefix':
                if (inputValue.match(/^\d{1,3}?$/)) {
                    this.setState({ [inputName]: inputValue });
                }
                return;
            case 'phoneLineNumber':
                if (inputValue.match(/^\d{1,4}?$/)) {
                    this.setState({ [inputName]: inputValue });
                }
                return;
            default:
                if (inputValue.match(/^\d{1,5}?$/)) {
                    this.setState({ [inputName]: inputValue });
                }
                return;
        }
    }

    handleSubmit = () => {
        const submitObject = Object.assign({}, this.state);
        this.props.onSubmit(submitObject);
    }

    render() {
        return (
            <div id="user_contact_info_widget">
                <div className="user_email_content" >
                    <div className="user_contact_info_content_title">
                        Email
                    </div>
                    <div className="user_email_input">
                        <input
                            readOnly={this.props.isReadOnly}
                            type="text"
                            name="email"
                            placeholder="Email"
                            className="text_input"
                            value={this.state.email}
                            onChange={this.handleEmailInput}
                            onBlur={this.handleEmailValidation}
                        />
                    </div>
                </div>
                <div className="user_phone_number_content">
                    <div className="user_contact_info_content_title">
                        Phone Number
                    </div>
                    <div className="user_phone_number_inputs">
                        <div className="user_phone_number_input">
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="phoneAreaCode"
                                placeholder="(919)"
                                className="text_input"
                                value={this.state.phoneAreaCode}
                                onChange={this.handlePhoneNumberValidation}
                            />
                        </div>
                        <div className="user_phone_number_input">
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="phonePrefix"
                                placeholder="123"
                                className="text_input"
                                value={this.state.phonePrefix}
                                onChange={this.handlePhoneNumberValidation}
                            />
                        </div>
                        <div className="user_phone_number_input">
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="phoneLineNumber"
                                placeholder="4567"
                                className="text_input"
                                value={this.state.phoneLineNumber}
                                onChange={this.handlePhoneNumberValidation}
                            />
                        </div>
                        <div className="user_phone_number_input">
                            <input
                                readOnly={this.props.isReadOnly}
                                type="text"
                                name="phoneExt"
                                placeholder="Ext."
                                className="text_input"
                                value={this.state.phoneExt}
                                onChange={this.handlePhoneNumberValidation}
                            />
                        </div>
                    </div>
                </div>
                <div className="user_contact_info_widget_button">
                    {!this.props.isReadOnly &&
                        <button
                            disabled={!this.state.isEmailValid}
                            onClick={this.handleSubmit}
                            className="button"
                        >
                            Update Contact Info
                    </button>
                    }
                </div>
            </div>
        )
    }
}

UserNameWidget.propTypes = {
    isReadOnly: PropTypes.bool,
    user: PropTypes.object.isRequired,
    onSubmit: PropTypes.func.isRequired,
};
