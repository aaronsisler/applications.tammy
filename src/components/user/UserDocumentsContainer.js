import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DocumentUploadWidget from '../shared/document/DocumentUploadWidget';
import DocumentList from '../shared/document/DocumentList';
import LoadingPage from '../core/LoadingPage';

export class UserDocumentsContainer extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
    }

    render() {
        return (
            <div>
                {!this.props.documents && <LoadingPage />}
                {this.props.documents &&
                    <div id="user_documents_container">
                        <DocumentUploadWidget />
                        <DocumentList
                            documents={this.props.documents}
                        />
                    </div>
                }
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    documents: state.userDocuments,
})

export default connect(mapStateToProps)(UserDocumentsContainer);

UserDocumentsContainer.propTypes = {
    documents: PropTypes.array.isRequired,
};
