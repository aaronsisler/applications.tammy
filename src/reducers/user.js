const userReducerDefaultState = null;

const userReducer = (state = userReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_USER':
            return action.user;
        case 'CLEAR_USER':
            return null;
        case 'EDIT_USER':
            return {
                ...state,
                ...action.updates
            }
        default:
            return state;
    }
};

export default userReducer;

