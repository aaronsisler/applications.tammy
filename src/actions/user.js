import database from 'Firebase/firebase';
import { setUser, editUser, clearUser } from './helpers/user';

export const startSetUser = () => (dispatch, getState) => {
    const { uid: userId } = getState().auth;

    return database.ref(`users/${userId}`).once('value').then((snapshot) => {
        dispatch(setUser(
            {
                userId: snapshot.key,
                ...snapshot.val()
            }));
    })
};

export const startEditUser = (updates) => (dispatch, getState) => {
    const { uid: userId } = getState().auth;

    return database.ref(`users/${userId}`).update(updates).then(() => {
        dispatch(editUser(updates));
    });
};

export const startClearUser = () => (dispatch) => dispatch(clearUser());
