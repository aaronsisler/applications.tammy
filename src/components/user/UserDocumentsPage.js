import React from 'react';
import UserDocumentsContainer from './UserDocumentsContainer';

export default class UserDocumentsPage extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div id="user_documents_page">
                <UserDocumentsContainer />
            </div>
        );
    }
}
