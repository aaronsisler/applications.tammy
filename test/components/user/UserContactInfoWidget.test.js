
import React from 'react';
import { shallow } from 'enzyme';
import UserContactInfoWidget from 'User/UserContactInfoWidget';
import user from '../../fixtures/user';
import InputTools from 'User/tools/inputs';
jest.mock('User/tools/inputs');

describe('UserContactInfoWidget', () => {
    const onSubmit = jest.fn();
    let wrapper;
    let instance;
    let inputToolsMock;

    const defaultEmptyState = {
        email: '',
        phoneAreaCode: '',
        phonePrefix: '',
        phoneLineNumber: '',
        phoneExt: '',
        isValidEmail: false
    };

    const buildWrapper = (userInput = {}, isReadOnly = true) => {
        wrapper = shallow(
            <UserContactInfoWidget
                isReadOnly={isReadOnly}
                onSubmit={onSubmit}
                user={userInput}
            />);
    };

    afterEach(() => {
        jest.restoreAllMocks();
    });

    it('should render correctly with default props', () => {
        buildWrapper();

        expect(wrapper.state()).toEqual(defaultEmptyState);
    });

    describe('when user props change', () => {
        const newEmail = 'new_email@taco.com';
        const newPhoneAreaCode = '123';
        const newPhonePrefix = '456';
        const newPhoneLineNumber = '7890';
        const newPhoneExt = '54321';

        const newUser = {
            email: newEmail,
            phoneAreaCode: newPhoneAreaCode,
            phonePrefix: newPhonePrefix,
            phoneLineNumber: newPhoneLineNumber,
            phoneExt: newPhoneExt,
        };

        beforeEach(() => {
            buildWrapper(user);
        });

        describe('when userId is the same', () => {
            it('should NOT update the state', () => {
                wrapper.setProps({
                    user: {
                        ...user,
                        ...newUser
                    }
                });

                expect(wrapper.state()).toEqual(
                    {
                        email: user.email,
                        phoneAreaCode: user.phoneAreaCode,
                        phonePrefix: user.phonePrefix,
                        phoneLineNumber: user.phoneLineNumber,
                        phoneExt: user.phoneExt,
                        isValidEmail: true
                    });
            });
        });

        describe('when userId is NOT the same', () => {
            it('should update the state', () => {
                wrapper.setProps(
                    {
                        user: {
                            ...newUser,
                            userId: 'Not Original Id'
                        }
                    });

                expect(wrapper.state()).toEqual({ ...newUser, isValidEmail: true });
            });

            it('should render correctly with default props', () => {
                wrapper.setProps(
                    {
                        user: {
                            userId: 'Not Original Id'
                        }
                    });

                expect(wrapper.state()).toEqual(defaultEmptyState);
            });
        });
    });

    describe('handleEmailValidation', () => {
        const name = 'email';
        const value = user.email;

        beforeEach(() => {
            InputTools.mockClear();
            buildWrapper(user, false);
            [inputToolsMock] = InputTools.mock.instances;
            instance = wrapper.instance();
        });

        describe('when email is valid', () => {
            it('should remove the required field classname from the email input element', () => {
                const target = { name, value };

                instance.handleEmailValidation({ target });

                expect(inputToolsMock.handleSetErrorClassname)
                    .toHaveBeenLastCalledWith(name);
            });

            it('should set isEmailValid in state to true', () => {
                const target = { name, value };

                instance.handleEmailValidation({ target });

                expect(wrapper.state('isValidEmail')).toBe(true);
            });
        });

        describe('when email is NOT valid', () => {
            const target = { name, value: 'bad_email' };
            it('should add the required field classname to the email input element', () => {
                instance.handleEmailValidation({ target });

                expect(inputToolsMock.handleSetErrorClassname)
                    .toHaveBeenLastCalledWith(name, false);
            });

            it('should set isEmailValid in state to false', () => {
                instance.handleEmailValidation({ target });

                expect(wrapper.state('isValidEmail')).toBe(false);
            });
        });
    });

    describe('when isReadOnly is FALSE', () => {
        beforeEach(() => {
            buildWrapper(user, false);
            instance = wrapper.instance();
        });

        it('should render UserContactInfoWidget correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        describe('Email Input', () => {
            const name = 'email';

            it('should set email on input change', () => {
                const value = 'jenny.appleseed@gmail.com';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(value);
            });

            xit('should call handleEmailValidation on blur', () => {
                jest.spyOn(instance, 'handleEmailValidation').mockImplementation(jest.fn());

                wrapper.find(`#${name}`).simulate('blur');

                expect(instance.handleEmailValidation).toHaveBeenCalled();
            });
        });

        describe('Phone Area Code Input', () => {
            const name = 'phoneAreaCode';

            it('should set phone area code on valid input change', () => {
                const value = '123';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should set phone area code to empty string when input is cleared', () => {
                const value = '';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should NOT set phone area code on invalid input change', () => {
                const value = 'a';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(user[name]);
            });
        });

        describe('Phone Prefix Input', () => {
            const name = 'phonePrefix';

            it('should set phone prefix on valid input change', () => {
                const value = '123';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should set phone prefix to empty string when input is cleared', () => {
                const value = '';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should NOT set phone prefix on invalid input change', () => {
                const value = 'a';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(user[name]);
            });
        });

        describe('Phone Line Number Input', () => {
            const name = 'phoneLineNumber';

            it('should set phone line number on valid input change', () => {
                const value = '1234';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should set phone line number to empty string when input is cleared', () => {
                const value = '';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should NOT set phone line number on invalid input change', () => {
                const value = 'a';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(user[name]);
            });
        });

        describe('Phone Ext Input', () => {
            const name = 'phoneExt';

            it('should set phone ext on valid input change', () => {
                const value = '01234';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should set phone ext to empty string when input is cleared', () => {
                const value = '';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(value);
            });

            it('should NOT set phone ext on invalid input change', () => {
                const value = 'a';
                const target = { name, value };

                wrapper.find(`#${name}`).simulate('change', { target });

                expect(wrapper.state(name)).toBe(user[name]);
            });
        });

        describe('Update Contact Info button', () => {
            it('should be enabled when isValidEmail is TRUE', () => {
                expect(wrapper.find('button').props().disabled).toBe(false);
            });

            it('should be disabled when isValidEmail is FALSE', () => {
                wrapper.setState({
                    email: 'taco.taco',
                    isValidEmail: false,
                });

                expect(wrapper.find('button').props().disabled).toBe(true);
            });

            it('should call onSubmit prop on button click', () => {
                const {
                    email,
                    phoneAreaCode,
                    phonePrefix,
                    phoneLineNumber,
                    phoneExt,
                } = user;

                wrapper.find('button').simulate('click');

                expect(onSubmit).toHaveBeenLastCalledWith({
                    email,
                    phoneAreaCode,
                    phonePrefix,
                    phoneLineNumber,
                    phoneExt,
                });
            });
        });
    });

    describe('when isReadOnly is TRUE', () => {
        it('should render UserContactInfoWidget correctly', () => {
            buildWrapper(user);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
