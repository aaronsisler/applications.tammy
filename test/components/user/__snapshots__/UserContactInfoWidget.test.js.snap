// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserContactInfoWidget when isReadOnly is FALSE should render UserContactInfoWidget correctly 1`] = `
<div
  id="user_contact_info_widget"
>
  <div
    className="user_email_content"
  >
    <div
      className="user_contact_info_content_title"
    >
      Email
    </div>
    <div
      className="user_email_input"
    >
      <input
        className="text_input"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email"
        readOnly={false}
        type="text"
        value="johnny.appleseed@gmail.com"
      />
    </div>
  </div>
  <div
    className="user_phone_number_content"
  >
    <div
      className="user_contact_info_content_title"
    >
      Phone Number
    </div>
    <div
      className="user_phone_number_inputs"
    >
      <div
        className="user_phone_number_input"
      >
        <input
          className="text_input"
          id="phoneAreaCode"
          name="phoneAreaCode"
          onChange={[Function]}
          placeholder="(919)"
          readOnly={false}
          type="text"
          value="919"
        />
      </div>
      <div
        className="user_phone_number_input"
      >
        <input
          className="text_input"
          id="phonePrefix"
          name="phonePrefix"
          onChange={[Function]}
          placeholder="123"
          readOnly={false}
          type="text"
          value="123"
        />
      </div>
      <div
        className="user_phone_number_input"
      >
        <input
          className="text_input"
          id="phoneLineNumber"
          name="phoneLineNumber"
          onChange={[Function]}
          placeholder="4567"
          readOnly={false}
          type="text"
          value="4567"
        />
      </div>
      <div
        className="user_phone_number_input"
      >
        <input
          className="text_input"
          id="phoneExt"
          name="phoneExt"
          onChange={[Function]}
          placeholder="Ext."
          readOnly={false}
          type="text"
          value="12345"
        />
      </div>
    </div>
  </div>
  <div
    className="user_contact_info_widget_button"
  >
    <button
      className="button"
      disabled={false}
      onClick={[Function]}
    >
      Update Contact Info
    </button>
  </div>
</div>
`;

exports[`UserContactInfoWidget when isReadOnly is TRUE should render UserContactInfoWidget correctly 1`] = `
<div
  id="user_contact_info_widget"
>
  <div
    className="user_email_content"
  >
    <div
      className="user_contact_info_content_title"
    >
      Email
    </div>
    <div
      className="user_email_input"
    >
      <input
        className="text_input"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email"
        readOnly={true}
        type="text"
        value="johnny.appleseed@gmail.com"
      />
    </div>
  </div>
  <div
    className="user_phone_number_content"
  >
    <div
      className="user_contact_info_content_title"
    >
      Phone Number
    </div>
    <div
      className="user_phone_number_inputs"
    >
      <div
        className="user_phone_number_input"
      >
        <input
          className="text_input"
          id="phoneAreaCode"
          name="phoneAreaCode"
          onChange={[Function]}
          placeholder="(919)"
          readOnly={true}
          type="text"
          value="919"
        />
      </div>
      <div
        className="user_phone_number_input"
      >
        <input
          className="text_input"
          id="phonePrefix"
          name="phonePrefix"
          onChange={[Function]}
          placeholder="123"
          readOnly={true}
          type="text"
          value="123"
        />
      </div>
      <div
        className="user_phone_number_input"
      >
        <input
          className="text_input"
          id="phoneLineNumber"
          name="phoneLineNumber"
          onChange={[Function]}
          placeholder="4567"
          readOnly={true}
          type="text"
          value="4567"
        />
      </div>
      <div
        className="user_phone_number_input"
      >
        <input
          className="text_input"
          id="phoneExt"
          name="phoneExt"
          onChange={[Function]}
          placeholder="Ext."
          readOnly={true}
          type="text"
          value="12345"
        />
      </div>
    </div>
  </div>
  <div
    className="user_contact_info_widget_button"
  />
</div>
`;
